text/0000-pack-cache-options.md

```diff
@@ -0,0 +1,222 @@
+
+The options available via this new flag would have sensible defaults based on the current behaiviour with the exception of the `--cache-image` which would be deprecated and later removed to reduce confusion. Overall UX would remain the same while allowing for greater flexibility.
+
+An additional benefit of this format is that it can later be expanded on if more options for a specific format were deamed necessary.
+
+```
+pack build --cache 'type=<type>;format=<format>;<additional-options>' --cache ...
+```
+
+### `cache` flag options
+* _**type**_ - The type of cache to configure [`build`, `launch`] (default: `build`).
+* _**format**_ - The format in which to store that cache [`dir`, `image`, `volume`] (default: `volume`). 
+
+###### Format specific options
+
+* `dir`
+    * _**path**_ - Path to the directory to use as cache.
+* `image`
+    * _**name**_ - Name of the image to use to save as cache (including tag).
+* `volume`
+    * _**name**_ - Name of the volume to use to save as cache. (default: `pack-<type>-<app-uuid>`)
+
+##### Examples
+
+```bash
+# specify named volume build cache
+pack build --cache 'type=build;format=volume;name=my-build-cache'
+
+# specify bind mount cache (relative to current working directory)
+pack build --cache 'type=build;format=dir;path=./my-build-cache'
+
+# specify named volume build cache (name only)
+pack build --cache 'name=my-cool-build-cache'
+
+# specify image cache
+pack build --cache 'type=build;format=image;name=io.docker.io/myorg/my-cache:build'
+
+# specify image cache (name only)
+pack build --cache 'format=image;name=io.docker.io/myorg/my-cache:build'
+```
+
+When no flags are provided the following equate to the defaults:
+
+```
+# default when no cache flag is provided
+pack build --cache 'type=build;format=volume;name=pack-<type>-<app-uuid>' --cache 'type=volume;format=volume;name=pack-<type>-<app-uuid>'
+```
+
+> NOTE: Templating as done internally by Pack for `<type>` and `<app-uuid>` is not a proposed part of this solution.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+<!-- This is the technical portion of the RFC, where you explain the design in sufficient detail.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work. -->
+
+Pack would parse the `--cache` flags and apply any options to the two particular cache types. It would take the necessary steps to provide the approprite configuration when executing the lifecycle.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+<!-- 
+Why should we *not* do this? -->
+
+#### Pros
+
+- The fomat is extensible to allow for additional cache formats or additional options.
+
+#### Cons
+
+- There is added complexity in the syntax since it's freeform.
+    - Given that the target use cases are more advanced this should be too impactful.
```

Comment:
Not saying it isn't worth it, but I don't think we should minimize the impact of creating complex custom usage. Ever since we introduce the buildpack URIs I find myself checking the docs frequently because I have trouble remembering the syntax, and I would consider myself an "advanced" user.

Reply:
Don't disagree but would you rather the alternative be used or is there another proposed option?

Reply:
Could we do a hybrid of this proposal and the listed alternative and remove `type` from this syntax. e.g. provide `--build-cache` and `--launch-cache` (or `--layers-cache` and `--daemon-cache`?) flags? I think this would allow `pack` to provide more meaningful help text for each type of cache and would allow us to more explicitly remove concepts like the launch cache, if we ever wanted to do that.

Reply:
@dwillist / @dfreilich what do y'all think about this proposal? I'm okay with `--build-cache` and `--launch-cache` and agree with the sentiment.